src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,55 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  margin: 40px auto 0;
+  max-width: 707px;
+
+  @each $day, $value in $start-days {
+    &--start-day-#{$day} {
+      .calendar__date:first-child {
+        margin-left: calc((100px + 1px) * #{$value});
+      }
+    }
+  }
+
+  @for $month-length from 28 through 31 {
+    &--month-length-#{$month-length} {
+      .calendar__date:nth-child(n + #{$month-length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__date {
+    position: relative;
+    box-sizing: border-box;
```

Comment:
Better is to move this property to * the selector and remove it from all other places.